WcaOnRails/app/models/competition.rb

```diff
@@ -824,6 +824,25 @@ def to_wcif
     }
   end
 
+  def set_wcif_events!(wcif_events)
+    # Remove extra events.
```

Comment:
> So when you do `competition_event.destroy!` here the foreign keys are taken care of?

So long as we remember to specify a [`dependent`](https://github.com/jfly/worldcubeassociation.org/blob/8bb76030faabbcf6efb77247372d5c2538b8f937/WcaOnRails/app/models/competition_event.rb#L7) clause on the association, yes!